packages/std-client/tsconfig.json

```diff
@@ -1,102 +1,18 @@
-    // "strictNullChecks": true,                         /* When type checking, take into account `null` and `undefined`. */
-    // "strictFunctionTypes": true,                      /* When assigning functions, check to ensure parameters and the return values are subtype-compatible. */
-    // "strictBindCallApply": true,                      /* Check that the arguments for `bind`, `call`, and `apply` methods match the original function. */
-    // "strictPropertyInitialization": true,             /* Check for class properties that are declared but not set in the constructor. */
-    // "noImplicitThis": true,                           /* Enable error reporting when `this` is given the type `any`. */
-    // "useUnknownInCatchVariables": true,               /* Type catch clause variables as 'unknown' instead of 'any'. */
-    // "alwaysStrict": true,                             /* Ensure 'use strict' is always emitted. */
-    // "noUnusedLocals": true,                           /* Enable error reporting when a local variables aren't read. */
-    // "noUnusedParameters": true,                       /* Raise an error when a function parameter isn't read */
-    // "exactOptionalPropertyTypes": true,               /* Interpret optional property types as written, rather than adding 'undefined'. */
-    // "noImplicitReturns": true,                        /* Enable error reporting for codepaths that do not explicitly return in a function. */
-    // "noFallthroughCasesInSwitch": true,               /* Enable error reporting for fallthrough cases in switch statements. */
-    // "noUncheckedIndexedAccess": true,                 /* Include 'undefined' in index signature results */
-    // "noImplicitOverride": true,                       /* Ensure overriding members in derived classes are marked with an override modifier. */
-    // "noPropertyAccessFromIndexSignature": true,       /* Enforces using indexed accessors for keys declared using an indexed type */
-    // "allowUnusedLabels": true,                        /* Disable error reporting for unused labels. */
-    // "allowUnreachableCode": true,                     /* Disable error reporting for unreachable code. */
-
-    /* Completeness */
-    // "skipDefaultLibCheck": true,                      /* Skip type checking .d.ts files that are included with TypeScript. */
-    "skipLibCheck": true /* Skip type checking all .d.ts files. */
+    "target": "es2020",
+    "module": "ES2022",
+    "moduleResolution": "node",
+    "resolveJsonModule": true,
+    "declaration": true,
+    "sourceMap": true,
+    "outDir": "dist",
+    "isolatedModules": true,
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "strict": true,
+    "skipLibCheck": true,
+    "jsx": "react-jsx"
   },
-  "exclude": ["dist", "**/*.spec.ts", "scripts", "packages"]
+  "exclude": ["dist"]
```

Comment:
looked into the test failures a bit, it seems they were more so revealed by this changed, then caused by it

```
  "include": ["src"],
  "exclude": ["**/*.spec.ts"]
```
I'd consider an explicit include like in some other packages, and a hacky exclude until specs stop causing typecheck errors

Reply:
I was able to fix these (my preference rather than ignoring) but now seeing

```
mud/packages/std-client % pnpm test

> @latticexyz/std-client@1.42.0 test /Users/kevin/Projects/latticexyz/mud/packages/std-client
> pnpm run build:mud && tsc --noEmit && vitest typecheck --run


> @latticexyz/std-client@1.42.0 build:mud /Users/kevin/Projects/latticexyz/mud/packages/std-client
> tsx ./scripts/tsgen.ts

Generated ts definition files: src/mud-definitions/store/contractComponents.ts
Generated ts definition files: src/mud-definitions/world/contractComponents.ts
../common/src/codegen/utils/format.ts:3:36 - error TS7016: Could not find a declaration file for module 'prettier-plugin-solidity'. '/Users/kevin/Projects/latticexyz/mud/node_modules/.pnpm/prettier-plugin-solidity@1.1.2_prettier@2.8.4/node_modules/prettier-plugin-solidity/src/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/prettier-plugin-solidity` if it exists or add a new declaration (.d.ts) file containing `declare module 'prettier-plugin-solidity';`

3 import prettierPluginSolidity from "prettier-plugin-solidity";
                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in ../common/src/codegen/utils/format.ts:3

 ELIFECYCLE  Test failed. See above for more details.
```

Reply:
that's even better, try copying `modules.d.ts` from another package